packages/bsky/src/data-plane/server/indexing/plugins/post.ts

```diff
@@ -444,10 +479,59 @@ async function validateReply(
   }
 }
 
+async function getThreadgateRecord(db: DatabaseSchema, postUri: string) {
+  const threadgateRecordUri = postUriToThreadgateUri(postUri)
+  const results = await db
+    .selectFrom('record')
+    .where('record.uri', '=', threadgateRecordUri)
+    .selectAll()
+    .execute()
+  const threadgateRecord = results.find(
+    (ref) => ref.uri === threadgateRecordUri,
+  )
+  if (threadgateRecord) {
+    return jsonStringToLex(threadgateRecord.json) as GateRecord
+  }
+}
+
+async function validatePostEmbed(
+  db: DatabaseSchema,
+  embedUri: string,
+  parentUri: string,
+) {
+  const postgateRecordUri = postUriToPostgateUri(embedUri)
+  const results = await db
+    .selectFrom('record')
+    .where('record.uri', '=', postgateRecordUri)
+    .selectAll()
+    .execute()
+  const postgateRecord = results.find((ref) => ref.uri === postgateRecordUri)
```

Comment:
Does this work just as well?
```suggestion
  const postgateRecord = await db
    .selectFrom('record')
    .where('record.uri', '=', postgateRecordUri)
    .selectAll()
    .executeTakeFirst()
```

Reply:
oh duh, yeah lgtm

Reply:
tested locally quick, committing